jdaviz/configs/imviz/plugins/tools.py

```diff
@@ -205,3 +205,33 @@ def on_mouse_or_key_event(self, data):
             with delay_callback(state, 'bias', 'contrast'):
                 state.bias = 0.5
                 state.contrast = 1
+
+
+@viewer_tool
+class SinglePixelRegion(CheckableTool):
+
+    icon = os.path.join(ICON_DIR, 'pixelspectra.svg')  # FIXME: Update icon
+    tool_id = 'jdaviz:singlepixelregion'
+    action_text = 'Create single-pixel spatial region'
+    tool_tip = 'Click on the viewer to create single-pixel spatial region'
+
+    def activate(self):
+        self.viewer.add_event_callback(self.on_mouse_event, events=['click'])
+
+    def deactivate(self):
+        self.viewer.remove_event_callback(self.on_mouse_event)
+
+    def on_mouse_event(self, data):
```

Comment:
@kecnry , other than the perceived behavior, I am not sure if there is much code in common with Cubeviz here.

Cubeviz has to support click to replace vs add new, but Imviz is just add new all the time.

Reply:
`apply_roi` is uglier. `load_regions` is well tested and is guaranteed to round-trip properly.

Reply:
ok, then can the new tool be in `core/tools.py`, and the cubeviz version inherit from that and just add the "alt" functionality on top?

Reply:
That is possible but then alt method can no longer just call the other method...

Reply:
OK refactored.